tasks/freeradius.yml

```diff
@@ -0,0 +1,170 @@
+    mode: 0640
+    owner: freerad
+    group: freerad
+  notify: restart freeradius
+
+- name: REST Configuration
+  when: openwisp2_radius
+  template:
+    src: freeradius/rest.j2
+    dest: "{{ freeradius_mods_available_dir }}/rest"
+    mode: 0640
+    owner: freerad
+    group: freerad
+  notify: restart freeradius
+
+- name: Enable REST module
+  when: openwisp2_radius
+  file:
+    src: "{{ freeradius_mods_available_dir }}/rest"
+    dest: "{{ freeradius_mods_enabled_dir }}/rest"
+    state: link
+    mode: 0640
+    owner: freerad
+    group: freerad
+
+- name: Remove default site
+  when: openwisp2_radius
+  file:
+    dest: "{{ freeradius_sites_enabled_dir }}/default"
+    state: absent
+
+- name: Ensure inner-tunnel site is present
+  when: openwisp2_radius
+  file:
+    src: "{{ freeradius_sites_available_dir }}/inner-tunnel"
+    dest: "{{ freeradius_sites_enabled_dir }}/inner-tunnel"
+    state: link
+    mode: 0640
+    owner: freerad
+    group: freerad
+
+- name: Captive portal configuration
+  when: openwisp2_radius
+  template:
+    src: freeradius/captiveportal.j2
+    dest: "{{ freeradius_sites_available_dir }}/captiveportal"
+    mode: 0640
+    owner: freerad
+    group: freerad
+  notify: restart freeradius
+
+- name: Captive portal to sites_enabled
+  when: openwisp2_radius
+  file:
+    src: "{{ freeradius_sites_available_dir }}/captiveportal"
+    dest: "{{ freeradius_sites_enabled_dir }}/captiveportal"
+    state: link
+    mode: 0640
+    owner: freerad
+    group: freerad
+  notify: restart freeradius
+  tags: [radius]
+
+# openwisp-radius cron jobs
+- name: delete_old_radacct
+  when: openwisp2_radius
+  cron:
+    name: delete_old_radacct
+    day: "*"
+    hour: 05
+    minute: 30
+    job: "/opt/openwisp2/env/bin/python /opt/openwisp2/manage.py delete_old_radacct 730"
+  tags: [openwisp2, radius]
+
+- name: delete_old_postauth
+  when: openwisp2_radius
+  cron:
+    name: delete_old_postauth
+    day: "*"
+    hour: 05
+    minute: 0
+    job: "/opt/openwisp2/env/bin/python /opt/openwisp2/manage.py delete_old_postauth 365"
+  tags: [openwisp2, radius]
+
+- name: cleanup_stale_radacct
+  when: openwisp2_radius
+  cron:
+    name: cleanup_stale_radacct
+    day: "*"
+    hour: 04
+    minute: 0
+    job: "/opt/openwisp2/env/bin/python /opt/openwisp2/manage.py cleanup_stale_radacct 1"
+  tags: [openwisp2, radius]
+
+- name: deactivate_expired_users
+  when: openwisp2_radius
+  cron:
+    name: deactivate_expired_users
+    day: "*"
+    hour: "*"
+    minute: "*/5"
+    job: "/opt/openwisp2/env/bin/python /opt/openwisp2/manage.py deactivate_expired_users"
+  tags: [openwisp2, radius]
+
+- name: delete_old_users
+  when: openwisp2_radius
+  cron:
+    name: delete_old_users
+    day: "*"
+    hour: "03"
+    minute: "30"
+    job: "/opt/openwisp2/env/bin/python /opt/openwisp2/manage.py delete_old_users"
+  tags: [openwisp2, radius]
```

Comment:
Can be simply removed by using celery-beat.

Reply:
cool, we should be able to add celery beat more easily once #206 and #215 are merged